code/datums/lazy_template.dm

```diff
@@ -74,15 +74,23 @@
 			z_lower = z_idx,
 		)
 		var/list/to_init = list()
+
+		// turfs can never be duplicated so keep them out of the below list to optimize O(n)
+		var/list/turfs = list()
+		// everything else that can be duplicated, such as areas or large objects
+		var/list/not_turfs = list()
```

Comment:
I mentioned areas because it'd be faster to keep them separate, since |= is O(N) and if we only need to do it a few times for each area that'll save time.
so basically do this but have a list for areas and one for movables